### DISCUSSION TITLE
Required information in assets metadata

### DISCUSSION THREAD

**Discussion by Alpenverein** [statamic/cms#5747] [Q&A]
Hi all
I am actually searching for a solution to the following problem:
Our assets have required information in the blueprint like "copyright" and "byline". But Users can upload assets and use this images aswell  without providing these information before.

Has anybody an idea how to solve this?

My ideas:
- If it was possible to filter the assets the assets-dialog shows, I could only list the images which have a copyright field filled out...
- If I was able to popup the asset-edit dialog after uploading an asset, the user would be ~forced~ remembered to fill out the missing information.
- Last idea: I could prevent showing images in the frontend/preview when information are not set...

**Comment by afonic** [comment]
An easy way to implement your last idea (if your asset field is called asset_field)

```
{{ asset_field }}
    {{ if copyright }}
        [ asset code here]
    {{ /if }}
{{ /asset_field }}
```

You can also make the fields required in the Blueprint, but that won't prevent them from uploading and using them.

There are probably more elegant solutions, depending on how much you use this, but to enforce this in the CP you probably need a custom fieldtype.

**Reply by afonic** [reply]
Actually another idea that we've used before is this: create a Grid field that has everything you need for each subfield. So you can have the asset it self and all other fields as required. This will work well if your fields aren't too many.

**Reply by Alpenverein** [reply]
Yeah, also a good idea... so a kind of wrapper around the asset..
